### PR TITLE
[Fix] Do not close spider with :itemcount_timeout when closespider_timeout is set to :disabled

### PR DESCRIPTION
**Pull request by maiphuong-van**
I notice that the spider is closed with reason "..itemcount timeout achieved" after 1 minute even though I do not config `closespider_timeout` (disabled by default). Oleg points out that it could be because in `Crawly.Manager.maybe_stop_spider_by_timeout/3` when limit is an atom, it is always evaluated to `true` and stop. I added a condition that the `limit` should be an integer, else it should go to the :ok clause.

### CODE REVIEW

File: test/manager_test.exs [hunk 1]
```diff
@@ -91,6 +90,20 @@ defmodule ManagerTest do
     assert %{} == Crawly.Engine.running_spiders()
   end
 
+  @tag timeout: 61_000
+  test "spider does not close after 1 minute when closespider timeout is disabled" do
+    Application.put_env(:crawly, :closespider_timeout, :disabled)
```
Comment (oltarasenko) [inline test/manager_test.exs:95]: @maiphuong-van What happens if you also use:
Application.put_env(:crawly, :manager_operations_timeout, 1_000)?

Comment (maiphuong-van) [reply test/manager_test.exs:95]: I can try that 👍

Comment (maiphuong-van) [reply test/manager_test.exs:95]: It works! I will push the update :)

File: test/manager_test.exs [hunk 2]
```diff
@@ -4,7 +4,6 @@ defmodule ManagerTest do
   setup do
     Application.put_env(:crawly, :concurrent_requests_per_domain, 1)
     Application.put_env(:crawly, :closespider_itemcount, 10)
-    Application.put_env(:crawly, :concurrent_requests_per_domain, 1)
```
Comment (maiphuong-van) [inline test/manager_test.exs:7]: Duplicate with line 5 so I removed it

**Review by oltarasenko — APPROVED** [review]

**Pull request merged**